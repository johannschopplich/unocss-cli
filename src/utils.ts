export function debouncePromise<T extends unknown[]>(
  fn: (...args: T) => Promise<void>,
  delay: number,
  onError: (err: unknown) => void
) {
  let timeout: ReturnType<typeof setTimeout> | undefined
  let promiseInFly: Promise<void> | undefined
  let callbackPending: (() => void) | undefined

  return function debounced(...args: Parameters<typeof fn>) {
    if (promiseInFly) {
      callbackPending = () => {
        debounced(...args)
        callbackPending = undefined
      }
    } else {
      if (timeout != null) clearTimeout(timeout)

      timeout = setTimeout(() => {
        timeout = undefined
        promiseInFly = fn(...args)
          .catch(onError)
          .finally(() => {
            promiseInFly = undefined
            if (callbackPending) callbackPending()
          })
      }, delay)
    }
  }
}
