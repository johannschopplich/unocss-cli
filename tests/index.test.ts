import { remove } from 'fs-extra'
import { run, cacheDir } from './utils'

beforeAll(async () => {
  await remove(cacheDir)
})

it('builds uno.css', async () => {
  const { output } = await run({
    'site/index.html': `<div class="p-4 max-w-screen-md"></div>`
  })

  expect(output).toMatchSnapshot()
})

it('supports unocss.config.js', async () => {
  const { output } = await run({
    'site/index.html': `<div class="box"></div>`,
    'unocss.config.js': `
      import { defineConfig } from 'unocss'
      export default defineConfig({
        shortcuts: [{ box: 'max-w-7xl mx-auto bg-gray-100 rounded-md shadow-sm p-4' }]
      })
    `
  })

  expect(output).toMatchSnapshot()
})
