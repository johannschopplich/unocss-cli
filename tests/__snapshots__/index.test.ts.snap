// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`builds uno.css 1`] = `
"/* generated by unocss-cli v0.2.0 */
/* layer: default */
.p-4{padding:1rem;}
.max-w-screen-md{max-width:768px;}"
`;

exports[`supports unocss.config.js 1`] = `
"/* generated by unocss-cli v0.2.0 */
/* layer: shortcuts */
.box{padding:1rem;margin-left:auto;margin-right:auto;--un-bg-opacity:1;background-color:rgba(244,244,245,var(--un-bg-opacity));border-radius:0.375rem;--un-shadow-color:0,0,0;--un-shadow:0 1px 2px 0 rgba(var(--un-shadow-color), 0.05);-webkit-box-shadow:var(--un-ring-offset-shadow, 0 0 #0000), var(--un-ring-shadow, 0 0 #0000), var(--un-shadow);box-shadow:var(--un-ring-offset-shadow, 0 0 #0000), var(--un-ring-shadow, 0 0 #0000), var(--un-shadow);max-width:7xl;}"
`;
